### ISSUE TITLE
/rotation/ gives a 500

### ISSUE THREAD

**Issue by vorpal-buildbot** [PennyDreadfulMTG/Penny-Dreadful-Tools#6547]
Reported on Discord by bakert#2193

**Comment by bakert** [comment]
```
Sep 29 05:27:35 katelyngigante uwsgi[832]: [2019-09-29 05:27:35,908] ERROR in app: Exception on /rotation/ [GET]
Sep 29 05:27:35 katelyngigante uwsgi[832]: Traceback (most recent call last):
Sep 29 05:27:35 katelyngigante uwsgi[832]:   File "/home/discord/.local/lib/python3.6/site-packages/flask/app.py", line 2446, in wsgi_app
Sep 29 05:27:35 katelyngigante uwsgi[832]:     response = self.full_dispatch_request()
Sep 29 05:27:35 katelyngigante uwsgi[832]:   File "/home/discord/.local/lib/python3.6/site-packages/flask/app.py", line 1951, in full_dispatch_request
Sep 29 05:27:35 katelyngigante uwsgi[832]:     rv = self.handle_user_exception(e)
Sep 29 05:27:35 katelyngigante uwsgi[832]:   File "/home/discord/.local/lib/python3.6/site-packages/flask/app.py", line 1820, in handle_user_exception
Sep 29 05:27:35 katelyngigante uwsgi[832]:     reraise(exc_type, exc_value, tb)
Sep 29 05:27:35 katelyngigante uwsgi[832]:   File "/home/discord/.local/lib/python3.6/site-packages/flask/_compat.py", line 39, in reraise
Sep 29 05:27:35 katelyngigante uwsgi[832]:     raise value
Sep 29 05:27:35 katelyngigante uwsgi[832]:   File "/home/discord/.local/lib/python3.6/site-packages/flask/app.py", line 1949, in full_dispatch_request
Sep 29 05:27:35 katelyngigante uwsgi[832]:     rv = self.dispatch_request()
Sep 29 05:27:35 katelyngigante uwsgi[832]:   File "/home/discord/.local/lib/python3.6/site-packages/flask/app.py", line 1935, in dispatch_request
Sep 29 05:27:35 katelyngigante uwsgi[832]:     return self.view_functions[rule.endpoint](**req.view_args)
Sep 29 05:27:35 katelyngigante uwsgi[832]:   File "./decksite/cache.py", line 80, in decorated_function
Sep 29 05:27:35 katelyngigante uwsgi[832]:     response = make_response(f(*args, **kwargs))
Sep 29 05:27:35 katelyngigante uwsgi[832]:   File "./decksite/main.py", line 298, in rotation
Sep 29 05:27:35 katelyngigante uwsgi[832]:     view = Rotation(interestingness, query)
Sep 29 05:27:35 katelyngigante uwsgi[832]:   File "./decksite/views/rotation.py", line 30, in __init__
Sep 29 05:27:35 katelyngigante uwsgi[832]:     self.runs, self.runs_percent, self.cards = rotation.read_rotation_files()
Sep 29 05:27:35 katelyngigante uwsgi[832]:   File "./magic/rotation.py", line 216, in read_rotation_files
Sep 29 05:27:35 katelyngigante uwsgi[832]:     c = process_score(name, hits, cs, runs, latest_list)
Sep 29 05:27:35 katelyngigante uwsgi[832]:   File "./magic/rotation.py", line 234, in process_score
Sep 29 05:27:35 katelyngigante uwsgi[832]:     c = cs[name]
Sep 29 05:27:35 katelyngigante uwsgi[832]: KeyError: 'Faerie Guidemother'
Sep 29 05:27:35 katelyngigante uwsgi[832]: [2019-09-29 05:27:35,910] ERROR in logger: At request path: /rotation/
Sep 29 05:27:35 katelyngigante uwsgi[832]: werkzeug.exceptions.InternalServerError: 500 Internal Server Error: The server encountered an internal error and was unable to complete your request. Either the server is overloaded or there is an error in the application.
Sep 29 05:27:35 katelyngigante uwsgi[832]: 500 error at /rotation/
Sep 29 05:27:35 katelyngigante uwsgi[832]:  500 Internal Server Error: The server encountered an internal error and was unable to complete your request. Either the server is overloaded or there is an error in the application.
Sep 29 05:27:35 katelyngigante uwsgi[832]: Reported on decksite by bakert99
Sep 29 05:27:35 katelyngigante uwsgi[832]: --------------------------------------------------------------------------------
Sep 29 05:27:35 katelyngigante uwsgi[832]: <details><summary><strong>Request Data</strong></summary>
Sep 29 05:27:35 katelyngigante uwsgi[832]: ```
Sep 29 05:27:35 katelyngigante uwsgi[832]: Request Method: GET
Sep 29 05:27:35 katelyngigante uwsgi[832]: Path: /rotation/?
Sep 29 05:27:35 katelyngigante uwsgi[832]: Cookies: {'__cfduid': 'd33ea97fcdcae1fa2a13fa1ba7a1662781561762006', '_ga': 'GA1.2.1963997268.1561762012', 'deck_id': '61212', 'session': '.eJx9UctuwjAQ_BefIxTn4STcGlFaiooAlT5VRW6yeTSxDY7TFhD_3g0SIPXQm2d3Z2Z3vCfJGrTgEqQhQ6M7sAjPRCVPIANRCZWdYdWmSmdJhRVCI595XkQDO6SeZzvMJxb5pyOToquas1ajUt4AToMkPSoKyPptWiWP-tRzLSJMoZKuBS256Gc_eA3aRBEyFO9M6SSt4abv3IvV09TWS1M7u9xV859YbuLyRl6NnA2fXeaNqtFwuCc8TaFtTxjXe5azKFvwOztawLRs8kkJ5uVhVotxvUM-_KwrDW3CMSnqM8cN_CAMB5Sx0KXupd-Hx2wvtG2LaMixVJ5NxsG29PPbR8pfR19TbmIW7FI2X12vlt-TGk0w3jVe84Y54pdU-RZrx9Ta82PwqdDi3SJH0cRsewKJgWvQ5GCRvxEargvAnWXXNIdf1cegXw.EG0OnA.qj5BDdQ6Q17DL_4BV2St6rTAVAA'}
Sep 29 05:27:35 katelyngigante uwsgi[832]: Endpoint: rotation
Sep 29 05:27:35 katelyngigante uwsgi[832]: View Args: {}
Sep 29 05:27:35 katelyngigante uwsgi[832]: Person: 195644917081440265
Sep 29 05:27:35 katelyngigante uwsgi[832]: Referrer: None
Sep 29 05:27:35 katelyngigante uwsgi[832]: Request Data: {}
Sep 29 05:27:35 katelyngigante uwsgi[832]: Host: pennydreadfulmagic.com
Sep 29 05:27:35 katelyngigante uwsgi[832]: Accept-Encoding: gzip
Sep 29 05:27:35 katelyngigante uwsgi[832]: Cf-Ipcountry: US
Sep 29 05:27:35 katelyngigante uwsgi[832]: X-Forwarded-For: 12.245.223.10, 172.68.143.176
Sep 29 05:27:35 katelyngigante uwsgi[832]: Cf-Ray: 51d81d80bb01ba10-SJC
Sep 29 05:27:35 katelyngigante uwsgi[832]: X-Forwarded-Proto: https
Sep 29 05:27:35 katelyngigante uwsgi[832]: Cf-Visitor: {"scheme":"https"}
Sep 29 05:27:35 katelyngigante uwsgi[832]: Upgrade-Insecure-Requests: 1
Sep 29 05:27:35 katelyngigante uwsgi[832]: User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/77.0.3865.90 Safari/537.36
Sep 29 05:27:35 katelyngigante uwsgi[832]: Sec-Fetch-Mode: navigate
Sep 29 05:27:35 katelyngigante uwsgi[832]: Sec-Fetch-User: ?1
Sep 29 05:27:35 katelyngigante uwsgi[832]: Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3
Sep 29 05:27:35 katelyngigante uwsgi[832]: Sec-Fetch-Site: none
Sep 29 05:27:35 katelyngigante uwsgi[832]: Accept-Language: en-US,en;q=0.9,es;q=0.8
Sep 29 05:27:35 katelyngigante uwsgi[832]: Cookie: __cfduid=d33ea97fcdcae1fa2a13fa1ba7a1662781561762006; _ga=GA1.2.1963997268.1561762012; deck_id=61212; session=.eJx9UctuwjAQ_BefIxTn4STcGlFaiooAlT5VRW6yeTSxDY7TFhD_3g0SIPXQm2d3Z2Z3vCfJGrTgEqQhQ6M7sAjPRCVPIANRCZWdYdWmSmdJhRVCI595XkQDO6SeZzvMJxb5pyOToquas1ajUt4AToMkPSoKyPptWiWP-tRzLSJMoZKuBS256Gc_eA3aRBEyFO9M6SSt4abv3IvV09TWS1M7u9xV859YbuLyRl6NnA2fXeaNqtFwuCc8TaFtTxjXe5azKFvwOztawLRs8kkJ5uVhVotxvUM-_KwrDW3CMSnqM8cN_CAMB5Sx0KXupd-Hx2wvtG2LaMixVJ5NxsG29PPbR8pfR19TbmIW7FI2X12vlt-TGk0w3jVe84Y54pdU-RZrx9Ta82PwqdDi3SJH0cRsewKJgWvQ5GCRvxEargvAnWXXNIdf1cegXw.EG0OnA.qj5BDdQ6Q17DL_4BV2St6rTAVAA
Sep 29 05:27:35 katelyngigante uwsgi[832]: Cf-Connecting-Ip: 12.245.223.10
Sep 29 05:27:35 katelyngigante uwsgi[832]: Cdn-Loop: cloudflare
Sep 29 05:27:35 katelyngigante uwsgi[832]: X-Forwarded-Host: pennydreadfulmagic.com
Sep 29 05:27:35 katelyngigante uwsgi[832]: X-Forwarded-Server: pennydreadfulmagic.com
Sep 29 05:27:35 katelyngigante uwsgi[832]: Connection: Keep-Alive
Sep 29 05:27:35 katelyngigante uwsgi[832]: ```
Sep 29 05:27:35 katelyngigante uwsgi[832]: </details>
Sep 29 05:27:35 katelyngigante uwsgi[832]: --------------------------------------------------------------------------------
Sep 29 05:27:35 katelyngigante uwsgi[832]: <details><summary>
Sep 29 05:27:35 katelyngigante uwsgi[832]: InternalServerError
Sep 29 05:27:35 katelyngigante uwsgi[832]: 500 Internal Server Error: The server encountered an internal error and was unable to complete your request. Either the server is overloaded or there is an error in the application.
Sep 29 05:27:35 katelyngigante uwsgi[832]: </summary>
Sep 29 05:27:35 katelyngigante uwsgi[832]: Stack Trace:
Sep 29 05:27:35 katelyngigante uwsgi[832]: ```Python traceback
Sep 29 05:27:35 katelyngigante uwsgi[832]:   File "/home/discord/.local/lib/python3.6/site-packages/flask/app.py", line 2463, in __call__
Sep 29 05:27:35 katelyngigante uwsgi[832]:     return self.wsgi_app(environ, start_response)
Sep 29 05:27:35 katelyngigante uwsgi[832]:   File "/home/discord/.local/lib/python3.6/site-packages/flask/app.py", line 2449, in wsgi_app
Sep 29 05:27:35 katelyngigante uwsgi[832]:     response = self.handle_exception(e)
Sep 29 05:27:35 katelyngigante uwsgi[832]: ```
Sep 29 05:27:35 katelyngigante uwsgi[832]: </details>
Sep 29 05:27:35 katelyngigante uwsgi[832]: Exception_hash: 59f0c756f815c99b42243d2c07a2549b5202ed12
Sep 29 05:27:36 katelyngigante uwsgi[832]: [2019-09-29 05:27:36,627] ERROR in logger: Github error
Sep 29 05:27:36 katelyngigante uwsgi[832]: 500 Internal Server Error: The server encountered an internal error and was unable to complete your request. Either the server is overloaded or there is an error in the application.
```

**Comment by bakert** [comment]
I put in a try...except here which stops the error but leaving the ticket open because that feels a bit risky. What if the cards causing the error are supposed to be in the rotation?

**Comment by bakert** [comment]
I suspect the underlying issue here is that we are not adding Adventures to the db correctly because they are kinda like split cards and kind of not.

**Comment by bakert** [comment]
OK adventures DO go into the db correctly. 

```
MariaDB [cards]> SELECT * FROM face WHERE name  = 'Faerie Guidemother';
+---------+--------------------+-----------+------+-------+-----------+---------+---------------------+-------------+------------+------+------+---------+----------+---------+
| id      | name               | mana_cost | cmc  | power | toughness | loyalty | type_line           | oracle_text | image_name | hand | life | starter | position | card_id |
+---------+--------------------+-----------+------+-------+-----------+---------+---------------------+-------------+------------+------+------+---------+----------+---------+
| 2683113 | Faerie Guidemother | {W}       |    1 | 1     | 1         | NULL    | Creature — Faerie   | Flying      | NULL       | NULL | NULL |    NULL | 1        |     123 |
+---------+--------------------+-----------+------+-------+-----------+---------+---------------------+-------------+------------+------+------+---------+----------+---------+
1 row in set (0.07 sec)
```

So it's a question of a restart only so the in-memory cache of the website picks it up? Is that something we can fire in the except?

**Comment by bakert** [comment]
OK so the issue was we had `Faerie Guidemother // Gift of the Fae` as the name.

Fixed that in 003edf0.

Leaving this open to revert the error-handling code in /rotation/. I *think* I prefer to explode there as it is an error to get a price for a card that "doesn't exist" but maybe we need another form of error handling that is "don't 500 but do send this message to mods"?

**Comment by bakert** [comment]
1c37c2e3 is the revert.
